cmd/encryption-v1.go

```diff
@@ -155,8 +153,6 @@ func rotateKey(oldKey []byte, newKey []byte, bucket, object string, metadata map
 }
 
 func newEncryptMetadata(key []byte, bucket, object string, metadata map[string]string, sseS3 bool) ([]byte, error) {
-	delete(metadata, crypto.SSECKey) // make sure we do not save the key by accident
```

Comment:
why are you removing htis?

Reply:
To make sure that it is not saved. At the moment there is no policy about this - see https://github.com/minio/minio/pull/6489#discussion_r219680647

Reply:
I mean why is the code removed .

Reply:
> I mean why is the code removed .

So the current code removes the key one it is consumed once, but in CopyObject() the key will be consumed twice. In GetObjectNInfo and then in rotateKey(), hence rotateKey() won't work without having to remove the key from the header when we consume it.

So, since we are using GetObjectNInfo() in CopyObject() now, 
Because in CopyObject(), crypto.SSECopyKey is consumed by GetObjectNInfo() before rotateKey

Reply:
Is there a written rule on what should be removed ? If not can we document it

Reply:
I included your new API in my PR @aead

Reply:
@vadmeste Can you actually push these changes?

Reply:
> @vadmeste Can you actually push these changes?

I already did @aead